### PR TITLE
[Supplier] Rename Supplier.Address to Supplier.OperatorAddress

### PR DESCRIPTION
**Pull request by red-0ne**
## Summary

This PR renames all the `Supplier.Address` references to `Supplier.OperatorAddress`.

It also include renaming `supplierAddress` to `supplierOperatorAddress` variables, methods in `Claim` and `Proof`, comments and error messages.

This PR has a lot of changes but does not introduce any addition logic to the code base.

## Issue

- #493 

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

**Documentation changes** (only if making doc changes)
- [ ] `make docusaurus_start`; only needed if you make doc changes

**Local Testing** (only if making code changes)
- [x] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- See [quickstart guide](https://dev.poktroll.com/developer_guide/quickstart) for instructions

**PR Testing** (only if making code changes)
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.
    - **THIS IS VERY EXPENSIVE**, so only do it after all the reviews are complete.
    - Optionally run `make trigger_ci` if you want to re-trigger tests without any code changes
    - If tests fail, try re-running failed tests only using the GitHub UI as shown [here](https://github.com/pokt-network/poktroll/assets/1892194/607984e9-0615-4569-9452-4c730190c1d2)

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [ ] I have commented my code
- [ ] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: pkg/client/interface.go
Comment (Olshansk) [inline pkg/client/interface.go:42]: If you do a non-case sensitive search for `supplieraddr` or `supplier addr` a few others come up.

I think it's a good reference to do a sanity check IF we need to add (for clarity) owner/operator there.

Screenshot 2024-08-07 at 1 02 15 PM
Screenshot 2024-08-07 at 1 02 10 PM

File: docusaurus/docs/protocol/primitives/claim_and_proof_lifecycle.md
Comment (Olshansk) [inline docusaurus/docs/protocol/primitives/claim_and_proof_lifecycle.md:725]: I did a ctrl-f for `supplier_addr` and found a few outstanding ones:

Screenshot 2024-08-07 at 12 59 33 PM

File: x/proof/module/query_claim.go
Comment (Olshansk) [inline x/proof/module/query_claim.go:73]: Can you add a `TODO_MAINNET: Move from Cobra to AutoCLI for all of these commands` please?

File: proto/poktroll/proof/tx.proto
Comment (Olshansk) [inline proto/poktroll/proof/tx.proto:77]: Nice. This is the most important part!

**Review by Olshansk — CHANGES_REQUESTED** [review]
Did not review everything, but I did a very simple ctrl-f all (left some comments) and took a look at the .proto files.

No issues, but just please do one more sanity check (e.g. update docs per my suggested search)

File: x/proof/module/query_claim.go
Comment (red-0ne) [reply x/proof/module/query_claim.go:73]: There's already `TODO_TECHDEBT`s in `x/<module>/module/tx.go which refer to all the `CmdXXX()`

`// TODO_TECHDEBT(@bryanchriswhite, #370): remove if custom query commands are consolidated into AutoCLI.`
 
Will change them to `TODO_MAINNET`

**Review by Olshansk — APPROVED** [review]
LGTM but let's merge this into `main` AFTER the others were merged as well.

**Pull request merged**
